ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -590,10 +590,30 @@ let rename (state : State.t)
   let open Fiber.O in
   let+ locs = Document.dispatch_exn doc command in
   let version = Document.version doc in
+  let source = Document.source doc in
   let edits =
-    List.map locs ~f:(fun loc ->
+    List.map locs ~f:(fun (loc : Warnings.loc) ->
         let range = Range.of_loc loc in
-        { TextEdit.newText = newName; range })
+        let make_edit () = TextEdit.create ~range ~newText:newName in
+        let occur_start_pos =
+          Position.of_lexical_position loc.loc_start |> Option.value_exn
+        in
+        match occur_start_pos with
+        | { character = 0; _ } -> make_edit ()
+        | pos -> (
+          let mpos = Position.logical pos in
+          let (`Offset index) = Msource.get_offset source mpos in
+          let source_txt = Msource.text source in
+          match source_txt.[index - 1] with
```

Comment:
Can `index` ever be 0?

Reply:
As far as I understand, in order for `index` to be equal to 0, `line` and `character` both need to be equal to 0, but I handle case of `line = _; character = 0` separately, so we should never get `index = 0`. 

I'll add `assert (index > 0)`.